code/modules/spells/spell_types/mind_transfer.dm

```diff
@@ -1,98 +1,126 @@
-*/
-/obj/effect/proc_holder/spell/targeted/mind_transfer/cast(list/targets, mob/living/user = usr, distanceoverride, silent = FALSE)
+/obj/effect/proc_holder/spell/pointed/mind_transfer/cast(list/targets, mob/living/user, distance_override = FALSE, silent = FALSE)
 	if(!targets.len)
 		if(!silent)
 			to_chat(user, "<span class='warning'>No mind found!</span>")
-		return
-
+		return FALSE
 	if(targets.len > 1)
 		if(!silent)
 			to_chat(user, "<span class='warning'>Too many minds! You're not a hive damnit!</span>")
-		return
+		return FALSE
 
 	var/mob/living/target = targets[1]
+	if(!swap_check(user, target, distance_override, silent))
+		return FALSE
+	if(istype(target, /mob/living/simple_animal/hostile/guardian))
+		var/mob/living/simple_animal/hostile/guardian/stand = target
+		if(stand.summoner)
+			target = stand.summoner
+
+	var/mob/living/victim = target //The target of the spell whos body will be transferred to.
+	var/mob/living/caster = user //The wizard/whomever doing the body transferring.
+
+	//MIND TRANSFER BEGIN
+	var/mob/dead/observer/ghost = victim.ghostize()
+	caster.mind.transfer_to(victim)
+
+	ghost.mind.transfer_to(caster)
+	if(ghost.key)
+		caster.key = ghost.key	//have to transfer the key since the mind was not active
+	qdel(ghost)
+	//MIND TRANSFER END
+
+	//Here we knock both mobs out for a time.
+	caster.Unconscious(unconscious_amount_caster)
+	victim.Unconscious(unconscious_amount_victim)
+	SEND_SOUND(caster, sound('sound/magic/mandswap.ogg'))
+	SEND_SOUND(victim, sound('sound/magic/mandswap.ogg')) // only the caster and victim hear the sounds, that way no one knows for sure if the swap happened
+	return TRUE
+
+/**
+  * swap_check: Checks if we are allowed and able to swap two minds
+  *
+  * Arguments:
+  * * user - caster of the spell
+  * * target - target of the spell
+  * * distance_override - if set to TRUE, range checks are ignored
+  * * silent - if set to TRUE, spell will not produce any warning messages to the user if the spell fails
+  */
+/obj/effect/proc_holder/spell/pointed/mind_transfer/proc/swap_check(mob/user, atom/target, distance_override = FALSE, silent = FALSE)
+	if(!isliving(target))
+		if(!silent)
+			to_chat(user, "<span class='warning'>You can only swap minds with living beings!</span>")
+		return FALSE
+	if(user == target)
+		if(!silent)
+			to_chat(user, "<span class='warning'>You can't swap minds with yourself!</span>")
+		return FALSE
 
-	var/t_He = target.p_they(TRUE)
-	var/t_is = target.p_are()
+	var/mob/living/victim = target
+	var/t_He = victim.p_they(TRUE)
+	var/t_is = victim.p_are()
 
-	if(!(target in oview(range)) && !distanceoverride)//If they are not in overview after selection. Do note that !() is necessary for in to work because ! takes precedence over it.
+	if(!(victim in oview(range)) && !distance_override) //If they are not in overview after selection. Do note that !() is necessary for in to work because ! takes precedence over it.

```

Comment:
The spell is now instant, is it not?
If so, this is no longer needed.

Reply:
Not sure I understand. Without the first check, the spell could be used at any range, even though it is supposed to be usable at 1 tile range only. The second check is for two cases where mind transfer is called directly from cast(), which are mass-swaps wizard events, without which it would not do anything either, as it would be constrained by range check.

Reply:
The way the spell used to work is that you casted it and it opened a list of targets where you would select yours from.
But the code had no guarantees on where the player would click okay on the list, so the `victim in oview(range)` was needed so they wouldn't be able to move around the corner and then selecting, for example.

But now there's no such delay, you click on a target and it happens, correct?

Reply:
That is correct, the spell happens instantly if conditions are met. Problem is that these conditions must include some kind of range check, because range varies spell by spell.

Reply:
Good news, I have came up with a more generic solution for all pointed spells, so this ugly specific check is no longer needed, more in the latest commit.